src/pathfinder/npf/npf.cpp

```diff
@@ -1040,10 +1040,18 @@ static NPFFoundTargetData NPFRouteInternal(AyStarNode *start1, bool ignore_start
 	/* Initialize user_data */
 	_npf_aystar.user_data = user;
 
+	/* We will limit the number of nodes for now, until we have a better
+	 * solution to really fix performance */
+	uint32_t max_search_nodes_before = _npf_aystar.max_search_nodes;
+	_npf_aystar.max_search_nodes = _settings_game.pf.npf.npf_max_search_nodes;
```

Comment:
This looks like it's doing what `AutoRestoreBackup` was designed/created for; temporarily setting a value to something different and then (automatically) reverting to the original state.

However, I doubt whether there's any use in restoring the value, as you'd be restoring it to the value when the game was loaded and this is the only function using it.